<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>865</width>
    <height>632</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Expert</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="minimumSize">
    <size>
     <width>633</width>
     <height>200</height>
    </size>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitH">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTableWidget" name="tableWidgetHistory">
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;历史文件信息&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="verticalHeaderMinimumSectionSize">
        <number>17</number>
       </attribute>
       <attribute name="verticalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string>Image</string>
        </property>
        <property name="textAlignment">
         <set>AlignLeading|AlignBottom</set>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Size</string>
        </property>
        <property name="textAlignment">
         <set>AlignLeading|AlignBottom</set>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Info</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>PathinWorkspace</string>
        </property>
       </column>
      </widget>
      <widget class="QScrollArea" name="scrollAreaMainDisplay">
       <property name="minimumSize">
        <size>
         <width>360</width>
         <height>240</height>
        </size>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>358</width>
          <height>300</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="CImageLabel" name="pictureBox">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="cursor">
            <cursorShape>CrossCursor</cursorShape>
           </property>
           <property name="mouseTracking">
            <bool>true</bool>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>865</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="tearOffEnabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>&amp;File</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <addaction name="action_Open"/>
    <addaction name="action_SaveAs"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menuImageProcessing">
    <property name="title">
     <string>&amp;ImageProcessing</string>
    </property>
    <widget class="QMenu" name="menuPointOperation">
     <property name="title">
      <string>Intensity Transformations </string>
     </property>
     <addaction name="actionInverse"/>
     <addaction name="actionLog_Transformations"/>
     <addaction name="actionGamma_Transformations"/>
     <addaction name="actionPiecewise_Liear_Transformation_Functions"/>
     <addaction name="separator"/>
     <addaction name="actionHistogram_Equalization"/>
     <addaction name="actionHistogram_Matching"/>
     <addaction name="actionLocal_Histogram_Processing"/>
     <addaction name="actionHistogram_Image_Enhancement"/>
     <addaction name="separator"/>
     <addaction name="actionGray_Strach"/>
     <addaction name="actionDifference"/>
     <addaction name="actionFusing"/>
     <addaction name="actionCorrelation"/>
     <addaction name="action_miu_delta_Equalization"/>
     <addaction name="actionBinary_Operation"/>
    </widget>
    <widget class="QMenu" name="menuSpatial_Filtering">
     <property name="title">
      <string>Spatial Filtering</string>
     </property>
     <addaction name="actionSmoothing_Spatial_Filters"/>
     <addaction name="actionSharpening_Spatial_Filters"/>
     <addaction name="actionImage_Enhance"/>
    </widget>
    <widget class="QMenu" name="menuFrequency_Domain">
     <property name="title">
      <string>Frequency Domain</string>
     </property>
     <addaction name="actionSmoothing"/>
     <addaction name="actionSharpening"/>
     <addaction name="actionSelective_Filtering"/>
    </widget>
    <widget class="QMenu" name="menuImage_Restoration_and_Reconstruction">
     <property name="title">
      <string>Color Image Processing</string>
     </property>
    </widget>
    <addaction name="menuPointOperation"/>
    <addaction name="actionNoise_Models"/>
    <addaction name="menuSpatial_Filtering"/>
    <addaction name="actiong_Fuzzy_Techniques"/>
    <addaction name="separator"/>
    <addaction name="menuFrequency_Domain"/>
    <addaction name="separator"/>
    <addaction name="menuImage_Restoration_and_Reconstruction"/>
    <addaction name="separator"/>
    <addaction name="actionMorphological_Image_Processing"/>
    <addaction name="separator"/>
    <addaction name="actionImage_Segmentation"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>&amp;Settings</string>
    </property>
    <widget class="QMenu" name="menuLanguages_2">
     <property name="title">
      <string>Languages</string>
     </property>
     <addaction name="actionChinese"/>
     <addaction name="actionEnglish_2"/>
    </widget>
    <addaction name="menuLanguages_2"/>
   </widget>
   <widget class="QMenu" name="menuVideoProcessing">
    <property name="title">
     <string>&amp;VideoProcessing</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_MachineLearning">
    <property name="title">
     <string>&amp;MachineLearning</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu3_D_Surfaces">
    <property name="title">
     <string>3&amp;D-Surfaces</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="actionFileAssistant"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menutest">
    <property name="title">
     <string>test</string>
    </property>
    <addaction name="actiontestFunction"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuImageProcessing"/>
   <addaction name="menuVideoProcessing"/>
   <addaction name="menu_MachineLearning"/>
   <addaction name="menu3_D_Surfaces"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_View"/>
   <addaction name="menuSettings"/>
   <addaction name="menutest"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Open"/>
   <addaction name="action_SaveAs"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidgetHistorgram">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Histogram</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsHistorgram">
    <property name="styleSheet">
     <string notr="true">border:1px solid gray;</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QComboBox" name="comboBoxHistorgram">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="currentText">
            <string/>
           </property>
           <property name="currentIndex">
            <number>-1</number>
           </property>
           <property name="maxVisibleItems">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="CImageLabel" name="labelHistorgram">
           <property name="minimumSize">
            <size>
             <width>128</width>
             <height>100</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>150</height>
            </size>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="pushButtonHistSaveImg">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>20</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="toolTip">
              <string>save current historgram as image</string>
             </property>
             <property name="text">
              <string>SaveImage</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButtonHistSaveData">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>20</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Save current historgram to file</string>
             </property>
             <property name="text">
              <string>SaveData</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetAppInfo">
   <property name="styleSheet">
    <string notr="true">*{border: 1px solid gray;}</string>
   </property>
   <property name="windowTitle">
    <string>Applicaton Log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QTableWidget" name="tableWidgetLogInfo">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>100</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>150</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="gridStyle">
        <enum>Qt::NoPen</enum>
       </property>
       <property name="columnCount">
        <number>3</number>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>10</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="verticalHeaderMinimumSectionSize">
        <number>17</number>
       </attribute>
       <column>
        <property name="text">
         <string>Time</string>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Command</string>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Arguments</string>
        </property>
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="styleSheet">
    <string notr="true">*{border: 1px solid gray;}</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>PushButton</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="styleSheet">
    <string notr="true">*{border: 1px solid gray;}</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4"/>
  </widget>
  <widget class="QDockWidget" name="dockWidgetVPHist">
   <property name="styleSheet">
    <string notr="true">*{border: 1px solid gray;}</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Vertical Project Historgram</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsVPHist">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>100</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">*{border: 1px solid gray;}</string>
       </property>
       <property name="text">
        <string>TextLabel</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/img/imgs/open.png</normaloff>:/img/imgs/open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="iconText">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open Image Files</string>
   </property>
   <property name="whatsThis">
    <string>Open Image Files</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
  </action>
  <action name="action_SaveAs">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/img/imgs/save.png</normaloff>:/img/imgs/save.png</iconset>
   </property>
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save As&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="whatsThis">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Exit&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="actionEnglish">
   <property name="text">
    <string>English</string>
   </property>
  </action>
  <action name="actionChinese">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Chinese</string>
   </property>
  </action>
  <action name="actionEnglish_2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>English</string>
   </property>
  </action>
  <action name="actionFileAssistant">
   <property name="text">
    <string>FileAssistant</string>
   </property>
  </action>
  <action name="actionInverse">
   <property name="text">
    <string>Negatives</string>
   </property>
  </action>
  <action name="actionLog_Transformations">
   <property name="text">
    <string>Log Transformations</string>
   </property>
  </action>
  <action name="actionGamma_Transformations">
   <property name="text">
    <string>Gamma Transformations</string>
   </property>
  </action>
  <action name="actionPiecewise_Liear_Transformation_Functions">
   <property name="text">
    <string>Piecewise-Liear Transformation Functions</string>
   </property>
  </action>
  <action name="actionHistogram_Equalization">
   <property name="text">
    <string>Histogram Equalization</string>
   </property>
  </action>
  <action name="actionHistogram_Matching">
   <property name="text">
    <string>Histogram Matching</string>
   </property>
  </action>
  <action name="actionLocal_Histogram_Processing">
   <property name="text">
    <string>Local Histogram Processing</string>
   </property>
  </action>
  <action name="actionHistogram_Image_Enhancement">
   <property name="text">
    <string>Histogram Image Enhancement</string>
   </property>
  </action>
  <action name="actionSmoothing_Spatial_Filters">
   <property name="text">
    <string>Smoothing Spatial Filters...</string>
   </property>
  </action>
  <action name="actionSharpening_Spatial_Filters">
   <property name="text">
    <string>Sharpening Spatial Filters...</string>
   </property>
  </action>
  <action name="actionGray_Strach">
   <property name="text">
    <string>Gray Strach</string>
   </property>
  </action>
  <action name="actionDifference">
   <property name="text">
    <string>Difference</string>
   </property>
  </action>
  <action name="actionFusing">
   <property name="text">
    <string>Fusing</string>
   </property>
  </action>
  <action name="actionCorrelation">
   <property name="text">
    <string>Correlation</string>
   </property>
  </action>
  <action name="action_miu_delta_Equalization">
   <property name="text">
    <string>(miu-delta)Equalization</string>
   </property>
  </action>
  <action name="actionBinary_Operation">
   <property name="text">
    <string>Binary Operation</string>
   </property>
  </action>
  <action name="actionImage_Enhance">
   <property name="text">
    <string>Image Enhancement</string>
   </property>
  </action>
  <action name="actiong_Fuzzy_Techniques">
   <property name="text">
    <string>Fuzzy Techniques</string>
   </property>
  </action>
  <action name="actionSmoothing">
   <property name="text">
    <string>Smoothing Filters</string>
   </property>
  </action>
  <action name="actionSharpening">
   <property name="text">
    <string>Sharpening Filters</string>
   </property>
  </action>
  <action name="actionSelective_Filtering">
   <property name="text">
    <string>Selective Filters</string>
   </property>
  </action>
  <action name="actionNoise_Models">
   <property name="text">
    <string>Noise Models</string>
   </property>
  </action>
  <action name="actionMorphological_Image_Processing">
   <property name="text">
    <string>Morphological Image Processing</string>
   </property>
  </action>
  <action name="actionImage_Segmentation">
   <property name="text">
    <string>Image Segmentation</string>
   </property>
  </action>
  <action name="actiontestFunction">
   <property name="text">
    <string>testFunction</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CImageLabel</class>
   <extends>QLabel</extends>
   <header location="global">cimagelabel.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
